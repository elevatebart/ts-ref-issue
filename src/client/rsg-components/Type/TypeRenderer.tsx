import React from "react";
import PropTypes from "prop-types";
import * as Rsg from "../../../typings";

export const styles = ({ fontFamily, fontSize, color }: Rsg.Theme) => ({
  type: {
    fontFamily: fontFamily.monospace,
    fontSize: fontSize.small,
    color: color.type
  }
});

interface TypeProps {
  children: React.ReactNode;
}

export const TypeRenderer: React.FunctionComponent<TypeProps> = ({
  children
}) => {
  return <span>{children}</span>;
};

TypeRenderer.propTypes = {
  children: PropTypes.node.isRequired
};

export default TypeRenderer;
