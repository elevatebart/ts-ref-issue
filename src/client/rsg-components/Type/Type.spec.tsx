import React from 'react';
import { shallow } from 'enzyme';
import { TypeRenderer, styles } from './TypeRenderer';

const props = {
	classes: classes(styles),
};

it('renderer should render type', () => {
	const actual = shallow(<TypeRenderer {...props}>Array</TypeRenderer>);

	expect(actual).toMatchSnapshot();
});
