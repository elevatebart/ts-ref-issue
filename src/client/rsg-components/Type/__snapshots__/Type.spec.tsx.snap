// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render type 1`] = `
<span
  className="type"
>
  Array
</span>
`;
